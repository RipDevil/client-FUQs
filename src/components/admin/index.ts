export * from './grips/users';
export * from './grips/fuqs';
export * from './portal';
