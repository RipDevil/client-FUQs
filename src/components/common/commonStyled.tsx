import styled from 'styled-components';

export const StyledSpinner = {
  SpinnerDiv: styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    z-index: 1000;
    margin: -50px 0px 0px -50px;
    max-width: 140px;
  `,
  BackgroundDiv: styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 999;
    background-color: rgb(255, 255, 255);
  `,
  TextDiv: styled.div`
    text-align: center;
    margin-top: 1rem;
  `,
};
