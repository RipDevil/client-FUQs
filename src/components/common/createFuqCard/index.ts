export { CreateFuqCard } from './create-fuq-card';
