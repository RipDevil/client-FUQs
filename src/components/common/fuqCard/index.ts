export { FuqCard } from './fuq-card';
