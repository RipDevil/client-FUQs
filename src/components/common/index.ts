export { Credits } from './credits';
export { FuqCard } from './fuqCard';
export { Spinner } from './spinner';
