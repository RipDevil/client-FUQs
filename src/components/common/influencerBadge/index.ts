export { InfluencerBadge } from './influencer-badge';
