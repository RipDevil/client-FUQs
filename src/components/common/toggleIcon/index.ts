export { ToggleIcon } from './toggle-icon';
