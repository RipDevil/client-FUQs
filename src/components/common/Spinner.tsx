import * as React from 'react';
import { StyledSpinner as S } from './commonStyled';

export interface SpinnerProps {
  text?: string;
  transparent?: boolean;
}

const Spinner: React.FC<SpinnerProps> = ({
  text = '',
  transparent = false, // TODO: implement this stuff
}) => {
  return (
    <>
      <S.BackgroundDiv />
      <S.SpinnerDiv>
        <div>
          <i className='fas fa-10x fa-spin fa-cog' />
        </div>
        {text && (
          <S.TextDiv>
            <h3>{text}</h3>
          </S.TextDiv>
        )}
      </S.SpinnerDiv>
    </>
  );
};

export default Spinner;
