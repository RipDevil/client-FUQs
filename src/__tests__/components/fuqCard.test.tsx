import { render } from '@testing-library/react';
import { FuqCard } from 'components/common';

test('FuqCard should be rendered', () => {
  const { getByText, container } = render(<FuqCard title="Some title" text="Some text" />);
  const [elementWithTitle, elementWithText] = [getByText('Some title'), getByText('Some text')];

  expect(elementWithTitle).toBeInTheDocument();
  expect(elementWithText).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
