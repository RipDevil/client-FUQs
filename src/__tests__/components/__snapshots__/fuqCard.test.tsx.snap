// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuqCard should be rendered 1`] = `
.c1 {
  font-size: 50px !important;
}

.c0 {
  cursor: pointer;
  font-size: 20px;
  overflow-y: hidden;
}

.c0:hover .ant-space-item {
  color: inherit;
}

.c0:hover .ant-badge-count {
  background: #ff4d4f;
}

.c0 .ant-space-item {
  -webkit-transition: 250ms ease;
  transition: 250ms ease;
  color: transparent;
}

.c0 .ant-badge-count {
  -webkit-transition: 250ms ease;
  transition: 250ms ease;
  background: none;
}

<div>
  <div
    class="ant-col c0"
  >
    <div
      class="ant-row ant-row-center"
    >
      <h1
        class="ant-typography ant-typography-danger ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line c1"
        title="Some title"
      >
        Some title
      </h1>
    </div>
    <div
      class="ant-row ant-row-center"
    >
      <h3
        class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-multiple-line"
        title="Some text"
      >
        Some text
      </h3>
    </div>
    <div
      class="ant-row ant-row-center"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
          style="margin-right: 8px;"
        >
          <span
            data-testid="icon-span"
          >
            <span>
              <span
                aria-label="like"
                class="anticon anticon-like"
                role="img"
                title=""
              >
                <svg
                  aria-hidden="true"
                  data-icon="like"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 00-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 00471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0142.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="ant-badge ant-badge-not-a-wrapper"
            >
              <sup
                class="ant-scroll-number ant-badge-count ant-badge-multiple-words"
                data-show="true"
                title="1817"
              >
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    1
                  </p>
                </span>
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    8
                  </p>
                </span>
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    1
                  </p>
                </span>
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    7
                  </p>
                </span>
              </sup>
            </span>
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          <span
            data-testid="icon-span"
          >
            <span>
              <span
                aria-label="fire"
                class="anticon anticon-fire"
                role="img"
                title=""
              >
                <svg
                  aria-hidden="true"
                  data-icon="fire"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M834.1 469.2A347.49 347.49 0 00751.2 354l-29.1-26.7a8.09 8.09 0 00-13 3.3l-13 37.3c-8.1 23.4-23 47.3-44.1 70.8-1.4 1.5-3 1.9-4.1 2-1.1.1-2.8-.1-4.3-1.5-1.4-1.2-2.1-3-2-4.8 3.7-60.2-14.3-128.1-53.7-202C555.3 171 510 123.1 453.4 89.7l-41.3-24.3c-5.4-3.2-12.3 1-12 7.3l2.2 48c1.5 32.8-2.3 61.8-11.3 85.9-11 29.5-26.8 56.9-47 81.5a295.64 295.64 0 01-47.5 46.1 352.6 352.6 0 00-100.3 121.5A347.75 347.75 0 00160 610c0 47.2 9.3 92.9 27.7 136a349.4 349.4 0 0075.5 110.9c32.4 32 70 57.2 111.9 74.7C418.5 949.8 464.5 959 512 959s93.5-9.2 136.9-27.3A348.6 348.6 0 00760.8 857c32.4-32 57.8-69.4 75.5-110.9a344.2 344.2 0 0027.7-136c0-48.8-10-96.2-29.9-140.9zM713 808.5c-53.7 53.2-125 82.4-201 82.4s-147.3-29.2-201-82.4c-53.5-53.1-83-123.5-83-198.4 0-43.5 9.8-85.2 29.1-124 18.8-37.9 46.8-71.8 80.8-97.9a349.6 349.6 0 0058.6-56.8c25-30.5 44.6-64.5 58.2-101a240 240 0 0012.1-46.5c24.1 22.2 44.3 49 61.2 80.4 33.4 62.6 48.8 118.3 45.8 165.7a74.01 74.01 0 0024.4 59.8 73.36 73.36 0 0053.4 18.8c19.7-1 37.8-9.7 51-24.4 13.3-14.9 24.8-30.1 34.4-45.6 14 17.9 25.7 37.4 35 58.4 15.9 35.8 24 73.9 24 113.1 0 74.9-29.5 145.4-83 198.4z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="ant-badge ant-badge-not-a-wrapper"
            >
              <sup
                class="ant-scroll-number ant-badge-count ant-badge-multiple-words"
                data-show="true"
                title="100"
              >
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    1
                  </p>
                </span>
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    0
                  </p>
                </span>
                <span
                  class="ant-scroll-number-only"
                  style="transition: none;"
                >
                  <p
                    class="ant-scroll-number-only-unit current"
                  >
                    0
                  </p>
                </span>
              </sup>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
